crates/ironrdp-rdpdr-native/src/nix/simple.rs

```diff
@@ -0,0 +1,765 @@
+            match std::fs::metadata(&file_full_path) {
+                Ok(meta) => {
+                    let file_attribute = get_file_attributes(&meta, file_name);
+                    if file_class == FileInformationClassLevel::FILE_BOTH_DIRECTORY_INFORMATION {
+                        let info = FileBothDirectoryInformation::new(
+                            transform_to_filetime(meta.ctime()),
+                            transform_to_filetime(meta.ctime()),
+                            transform_to_filetime(meta.atime()),
+                            transform_to_filetime(meta.mtime()),
+                            meta.size() as i64,
+                            file_attribute,
+                            file_name.to_string(),
+                        );
+                        let info2 = FileInformationClass::BothDirectory(info);
+                        Ok(vec![SvcMessage::from(RdpdrPdu::ClientDriveQueryDirectoryResponse(
+                            ClientDriveQueryDirectoryResponse {
+                                device_io_reply: DeviceIoResponse::new(device_io_request, NtStatus::SUCCESS),
+                                buffer: Some(info2),
+                            },
+                        ))])
+                    } else {
+                        warn!("unsupported file class for query directory");
+                        Ok(vec![SvcMessage::from(RdpdrPdu::ClientDriveQueryDirectoryResponse(
+                            ClientDriveQueryDirectoryResponse {
+                                device_io_reply: DeviceIoResponse::new(device_io_request, NtStatus::NOT_SUPPORTED),
+                                buffer: None,
+                            },
+                        ))])
+                    }
+                }
+                Err(e) => {
+                    warn!("metadata error:{}", e);
+                    Ok(vec![SvcMessage::from(RdpdrPdu::ClientDriveQueryDirectoryResponse(
+                        ClientDriveQueryDirectoryResponse {
+                            device_io_reply: DeviceIoResponse::new(device_io_request, not_found_status),
+                            buffer: None,
+                        },
+                    ))])
+                }
+            }
+        }
+    }
+}
+pub(crate) fn query_directory(
+    backend: &mut SimpleRdpdrBackend,
+    req_inner: ServerDriveQueryDirectoryRequest,
+) -> PduResult<Vec<SvcMessage>> {
+    match backend.file_path_map.get(&req_inner.device_io_request.file_id) {
+        Some(parent_pos_for_next) => {
+            let mut find_file_name = None;
+            if req_inner.initial_query > 0 {
+                if req_inner.path.ends_with('*') {
+                    unsafe {
+                        let mut parent = backend.file_base.clone();
+                        let query_path = req_inner.path.replace('\\', "/");
+                        let len = query_path.len();
+                        parent.push_str(&query_path[0..len - 1]);
```

Comment:
suggestion: FYI, we are using the `clippy::arithmetic_side_effects` to warn us about potential arithmetic side-effect bugs. Typically, we’ll `allow` it and explain why there is no problem with a comment. Alternatively, it’s possible to use checked arithmetic operations.

For instance:

https://github.com/Devolutions/IronRDP/blob/abc0ee0617fcdcda3f0ace3ffd292251979d1910/crates/ironrdp-cliprdr-format/src/bitmap.rs#L497-L501

https://github.com/Devolutions/IronRDP/blob/abc0ee0617fcdcda3f0ace3ffd292251979d1910/crates/ironrdp-web/src/session.rs#L548-L550

See also: https://github.com/Devolutions/IronRDP/blob/abc0ee0617fcdcda3f0ace3ffd292251979d1910/STYLE.md#explain-non-obvious-assumptions-by-referencing-the-invariants

Reply:
ok, it seems these warning will emit after i add clippy warns. And therefore i modify the code